safe_core/src/client/mod.rs

```diff
@@ -2200,11 +2180,8 @@ mod tests {
                         .ins(b"key3".to_vec(), b"value".to_vec(), 0);
 
                     client3
-                        .mutate_seq_mdata_entries(name, tag, entry_actions.clone())
-                        .then(move |res| {
-                            assert_eq!(unwrap!(res), ());
-                            Ok(())
-                        })
+                        .mutate_seq_mdata_entries(name, tag, entry_actions)
+                        .then(move |_| Ok(()))
```

Comment:
When using `.then(...)` if the result is ignored then we wouldn't know if there is a failure. `.and_then()` on the other hand would propagates the error to the outer function. However, asserting that the result is a success would help easily pinpoint which function failed. If you agree, could you please make this change here and in some other places too?
```suggestion
                        .then(move |res| {
                             assert!(res.is_ok());
                             Ok(())
                        })
```